src/typechecker.erl

```diff
@@ -3453,27 +3454,44 @@ pick_one_refinement_each([Ty|Tys], [RefTy|RefTys]) ->
     RefHeadCombinations ++ RefTailCombinations.
 
 %% Is a type refinable to the point that we do exhaustiveness checking on it?
-refinable(?type(integer)) ->
+refinable(TEnv, Ty) ->
+    refinable(TEnv, Ty, sets:new()).
+
+refinable(_TEnv, ?type(integer), _Trace) ->
```

Comment:
Perhaps a matter of taste, and I know it's not even consistent within this module, but I think `refinable(Ty, TEnv)` is a better order of the parameters.